substrate/frame/distribution/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+	pub(super) type CompletedSpendings<T: Config> =
+		StorageMap<_, Twox64Concat, SpendingIndex, SpendingInfo<T>, OptionQuery>;
+
+	/// Spendings that still have to be completed.
+	#[pallet::storage]
+	pub(super) type Spendings<T: Config> =
+		StorageMap<_, Twox64Concat, SpendingIndex, SpendingInfo<T>, OptionQuery>;
+
+	/// List of whitelisted projects to be rewarded
+	#[pallet::storage]
+	pub type Projects<T: Config> =
+		StorageValue<_, BoundedVec<ProjectInfo<T>, T::MaxProjects>, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Reward successfully claimed
+		RewardClaimed {
+			when: BlockNumberFor<T>,
+			amount: BalanceOf<T>,
+			project_account: ProjectId<T>,
+		},
+
+		/// A Spending was created
+		SpendingCreated {
+			when: BlockNumberFor<T>,
+			amount: BalanceOf<T>,
+			project_account: ProjectId<T>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Not enough Funds in the Pot
+		InsufficientPotReserves,
+		/// The funds transfer operation failed
+		TransferFailed,
+		/// Spending or spending index does not exists
+		InexistentSpending,
+		/// No valid Account_id found
+		NoValidAccount,
+		/// No project available for funding
+		NoProjectAvailable,
+		/// The Funds transfer failed
+		FailedSpendingOperation,
+		/// Still not in claiming period
+		NotClaimingPeriod,
+		/// Funds locking failed
+		FundsReserveFailed,
+		/// An invalid result  was returned
+		InvalidResult,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Weight: see `begin_block`
+		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
+			Self::begin_block(n)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// OPF Reward Claim logic
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// From this extrinsic any user can claim a reward for a nominated/whitelisted project.
+		///
+		/// ### Parameters
+		/// - `project_account`: The account that will receive the reward.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::InexistentSpending`]:Spending or spending index does not exists
+		/// - [`Error::<T>::NoValidAccount`]:  No valid Account_id found
+		/// - [`Error::<T>::NotClaimingPeriod`]: Still not in claiming period
+		///  
+		/// ## Events
+		/// Emits [`Event::<T>::RewardClaimed`] if successful for a positive approval.
+		#[pallet::call_index(0)]
+		pub fn claim_reward_for(
+			origin: OriginFor<T>,
+			project_account: ProjectId<T>,
+		) -> DispatchResult {
+			let _caller = ensure_signed(origin)?;
+			let spending_indexes = Self::get_spending(project_account);
+			let pot = Self::pot_account();
+			for i in spending_indexes {
+				let mut info = Spendings::<T>::get(i).ok_or(Error::<T>::InexistentSpending)?;
+				let project_account =
+					info.whitelisted_project.clone().ok_or(Error::<T>::NoValidAccount)?;
+				let now = <frame_system::Pallet<T>>::block_number();
```

Comment:
do not read block numbers directly and in this case, use time instead of block number
https://github.com/paritytech/polkadot-sdk/issues/3268

Reply:
This kind of new for me. can you point me to an updated pallet in particular please? 
I would like to see it used in context.

Reply:
most of the pallets are still pending to be updated but here is an example of how to use the `Time` trait to access time https://github.com/paritytech/polkadot-sdk/blob/776e95748901b50ff2833a7d27ea83fd91fbf9d1/substrate/frame/contracts/src/lib.rs#L258

Reply:
An easier example is the vesting pallet, that just pulls the block number from a trait:

https://github.com/paritytech/polkadot-sdk/blob/7ecf3f757a5d6f622309cea7f788e8a547a5dce8/substrate/frame/vesting/src/lib.rs#L183

Either way, the system block number is not good because parachains blocktimes are much less stable than relay block numbers.

Reply:
Corrected